bucket.go

```diff
@@ -433,22 +419,10 @@ type XattrStore interface {
 	// - opts: Options; use PreserveExpiry to avoid setting expiry
 	// - previous: The current document, if known. Will be used in place of the initial Get
 	// - callback: The callback that mutates the document
-	WriteUpdateWithXattr(ctx context.Context, k string, xattrKey string, userXattrKey string, exp uint32, previous *BucketDocument, opts *MutateInOptions, callback WriteUpdateWithXattrFunc) (casOut uint64, err error)
+	WriteUpdateWithXattrs(ctx context.Context, k string, xattrs []string, exp uint32, previous *BucketDocument, opts *MutateInOptions, callback WriteUpdateWithXattrsFunc) (casOut uint64, err error)
 
 	// Updates a document's xattr.
-	UpdateXattr(ctx context.Context, k string, xattrKey string, exp uint32, cas uint64, xv interface{}, opts *MutateInOptions) (casOut uint64, err error)
-
-	// Updates an xattr and deletes the body (making the doc a tombstone.) (?)
-	UpdateXattrDeleteBody(ctx context.Context, k, xattrKey string, exp uint32, cas uint64, xv interface{}, opts *MutateInOptions) (casOut uint64, err error)
-
-	// Deletes the document's body. Updates the CAS and CRC32 macros in the specified xattr.
-	DeleteBody(ctx context.Context, k string, xattrKey string, exp uint32, cas uint64, opts *MutateInOptions) (casOut uint64, err error)
```

Comment:
This used to be used in the case where the bucket doesn't support BucketStoreFeature CreateDeletedWithXattr, and we needed to go back and delete the body when inserting a new server tombstone w/ xattr. I don't see that handling in the updated createTombstone.  Is this intentional (i.e. we no longer support server versions without CreateDeletedWithXattr?).  If that were the case I'd expect the supportsTombstoneCreation check in createTombstone could be removed.

Reply:
I just took these functions out of sg-bucket because they are private to gocb implementation and made them private to sync gateway base package. It works correctly for server 6.5 -> 7.6 in the tests, but they could be missing parts.

In `WriteTombstoneWithXattrs`, we only need to delete the body if we had to create a server tombstone. This logic looks identical to me in the sync gateway PR as it does in the main branch.